src/shared/remoteSession.ts

```diff
@@ -173,13 +180,47 @@ export async function handleMissingTool(tools: Err<MissingTool[]>) {
 export async function getDeniedSsmActions(client: IamClient, roleArn: string): Promise<IAM.EvaluationResult[]> {
     const deniedActions = await client.getDeniedActions({
         PolicySourceArn: roleArn,
-        ActionNames: [
-            'ssmmessages:CreateControlChannel',
-            'ssmmessages:CreateDataChannel',
-            'ssmmessages:OpenControlChannel',
-            'ssmmessages:OpenDataChannel',
-        ],
+        ActionNames: minimumSsmActions,
     })
 
     return deniedActions
 }
+
+export async function promptToAddPolicies(client: IamClient, roleArn: string): Promise<boolean> {
+    const promptText = `In order to connect, toolkit will add actions ${getFormattedSsmActions()} to role ${roleArn}`
+    const confirmation = await showConfirmationMessage({ prompt: promptText, confirm: 'Approve' })
+
+    if (confirmation) {
+        await addSsmActionsToInlinePolicy(client, roleArn)
+    }
+
+    return confirmation
+}
+
+function getFormattedSsmActions() {
+    const formattedActions = minimumSsmActions.map(action => `${action},\n`).reduce((l, r) => l + r)
+
+    return formattedActions.slice(0, formattedActions.length - 2)
+}
+
+function getSsmPolicyDocument() {
+    return `{
+            \"Version\": \"2012-10-17\",
```

Comment:
is this escaping necessary?
```suggestion
            "Version": "2012-10-17",
```

Reply:
Oh, good catch. The example in the documentation made me think it was needed, but appears to work without. https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/IAM.html#putRolePolicy-property